### PR TITLE
[risk=no][RW-10063] Remove deprecated fields from leonardo.yaml

### PR DESCRIPTION
**Pull request by jmthibault79**
We may accidentally use fields which exist - so it's better to remove them entirely.

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/api/DisksControllerTest.java [hunk 1]
```diff
@@ -119,7 +119,10 @@ public void test_getDisk() throws ApiException {
             .diskType(LeonardoDiskType.STANDARD)
             .status(LeonardoDiskStatus.READY)
             .auditInfo(new LeonardoAuditInfo().createdDate(createDate).creator(user.getUsername()))
-            .googleProject(GOOGLE_PROJECT_ID);
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/DisksControllerTest.java:122]: missed this in #7668

**Comment by jmthibault79** [comment]
1 e2e failure: notebook-create-python.  May be relevant?  I'll rerun and also test manually.

**Comment by jmthibault79** [comment]
most recent e2e run had 1 failure too, but a different one: tests/notebook/notebook-writer-clone-workspace.spec.ts

So I'm calling this a flake.

**Comment by jmthibault79** [comment]
Tested locally.  Seems to work fine.

**Review by evrii — APPROVED** [review]
Looks good to me.

**Pull request merged**